src/triangle.rb

```diff
@@ -14,13 +14,15 @@
 #   about_triangle_project_2.rb
 #
 def triangle(a, b, c)
-  # WRITE THIS CODE
-  #--
-  a, b, c = [a, b, c].sort
-  fail TriangleError if (a+b) <= c
-  sides = [a, b, c].uniq
-  [nil, :equilateral, :isosceles, :scalene][sides.size]
-  #++
+  raise TriangleError.new if [a,b,c].include?(0)
+  raise TriangleError.new if [a,b,c].map(&:positive?).include?(false)
+
+  x, y, z = [a,b,c].sort
+  raise TriangleError if x + y <= z
+
+  return :equilateral if a == b && b == c
+  return :isosceles if a == b || a == c || b == c
+  return :scalene if a != b && a !=c && b != c
```

Comment:
No need to check last condition. Just return `:scalene` as last possible option
```suggestion

  :scalene
```